lib/src/ui/scaner_screen.dart

```diff
@@ -84,6 +86,8 @@ class ScanerScreenState extends State<ScanerScreen> {
                         .toList(),
                   ),
                 ),
+                if (bleScanner.state.scanIsInProgress)
+                  const CircularProgressIndicator(),
```

Comment:
Hmm, not sure if additional progress indicator doesn't overload UI. We know that scan in progress from incactive start and active stop button

Reply:
This indicate that something is happening and app is still alive. I would keep it. This animation should not be to complex to ui.

Reply:
For end user it additional element to look on

Reply:
Maybe `LinearProgressIndicator` at the top of the list will not overload interface to mach.

Reply:
Maybe three dots animation after last element/card of ListView.
Something like: https://pub.dev/packages/progress_indicators. Or even better: https://pub.dev/packages/dots_indicator

Reply:
I would keep current indicator (to not add external dependencies)